ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -314,7 +314,10 @@ forkBlockProduction maxBlockBodySize IS{..} BlockProduction{..} =
       trace $ TraceForgeAboutToLead currentSlot
       leaderResult <- withTxs $ \MempoolSnapshot{snapshotTxsForSize} -> do
         l@ExtLedgerState{..} <- ChainDB.getCurrentLedger chainDB
-        let txs = map fst (snapshotTxsForSize maxBlockBodySize)
+        let blockEncodingOverheadSize = 650  {- max block header size -}
+                                      + 8    {- block body encoding overhead -}
```

Comment:
The overhead depends on the `blk`, so this should also be a method in `RunNode`.

Also, it would be better to explain how you got to 8:
```haskell
let blockBodyOverhead = 1 {- ABody: encodeListLen 4 -}
                      + 1 {- TxPayload: list -} -- Or is it 2?
                      + ...
```